fabric-rendering-v1/src/main/java/net/fabricmc/fabric/impl/client/rendering/BuiltinItemRendererRegistryImpl.java

```diff
@@ -50,7 +59,23 @@ public void register(Item item, BuiltinItemRenderer renderer) {
 	}
 
 	/* @Nullable */
-	public static BuiltinItemRenderer getRenderer(Item item) {
+	public static DynamicItemRenderer getRenderer(Item item) {
 		return RENDERERS.get(item);
 	}
+
+	/**
+	 * An implementation of the new renderer that forwards calls to an old renderer.
+	 */
+	static class OldRendererImpl implements DynamicItemRenderer {
+		private final BuiltinItemRenderer renderer;
+
+		private OldRendererImpl(BuiltinItemRenderer renderer) {
+			this.renderer = renderer;
```

Comment:
can just convert this to a lambda imo since that is not performance-wise worse

Reply:
Just messy as a Lambda imo. But I could